### PR TITLE
feat: Implement onboarding step management on the flash

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements https://app.clickup.com/t/9002019994/PRF-4769

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4769 flash tlv implementation](https://app.clickup.com/t/9002019994/PRF-4769)

File: common/interfaces/flash_interface/flash_struct.h [hunk 1]
```diff
@@ -118,6 +118,7 @@ typedef struct Flash_Struct {
   uint8_t displayRotation;
   uint8_t enable_passphrase;
   uint8_t enable_log;
+  uint8_t onboarding_step;
```
Comment (ujjwal-cyph) [inline common/interfaces/flash_interface/flash_struct.h:121]: please update the MACRO definition `FLASH_STRUCT_TLV_SIZE` as we are adding a new member to `Flash_Struct`

File: apps/manager_app/onboarding/onboarding.c [hunk 2]
```diff
@@ -157,32 +155,35 @@ const flow_step_t *onboarding_get_step(void) {
...
-  // TODO: Get last_step from flash and update next_step in flash
-  /* Ensure we never go back a step */
-  if ((ONBOARDING_COMPLETE != last_step) && (last_step < next_step)) {
-    last_step = next_step;
+  onboarding_steps_e last_step = onboarding_get_last_step();
+
+  /* Check for DEFAULT_VALUE_IN_FLASH to save the state in a virgin device and
+   * ensure we never go back a step */
+  if ((DEFAULT_VALUE_IN_FLASH == last_step) ||
+      ((ONBOARDING_COMPLETE != last_step) && (last_step < next_step))) {
```
Comment (ujjwal-cyph) [inline apps/manager_app/onboarding/onboarding.c:172]: can simplify this check.

`(ONBOARDING_COMPLETE != last_step) && (last_step < next_step)` is redundant

Comment (kartikCypherock) [reply apps/manager_app/onboarding/onboarding.c]: I re-checked the logic and it does seem to be redundant. Updated the same. Thanks

**Review by ujjwal-cyph — APPROVED** [review]

**Comment by kartikCypherock** [comment]
Will be a part of https://github.com/Cypherock/x1_wallet_firmware/pull/276, so closing this PR

**Pull request closed without merging**